packages/replay-internal/src/coreHandlers/performanceObserver.ts

```diff
@@ -26,13 +37,18 @@ export function setupPerformanceObserver(replay: ReplayContainer): () => void {
   });
 
   clearCallbacks.push(
-    addLcpInstrumentationHandler(({ metric }) => {
-      replay.replayPerformanceEntries.push(getLargestContentfulPaint(metric));
-    }),
+    addLcpInstrumentationHandler(webVitalHandler(getLargestContentfulPaint)),
+    addClsInstrumentationHandler(webVitalHandler(getCumulativeLayoutShift)),
+    addFidInstrumentationHandler(webVitalHandler(getFirstInputDelay)),
+    addInpInstrumentationHandler(webVitalHandler(getInteractionToNextPaint)),
   );
 
   // A callback to cleanup all handlers
   return () => {
     clearCallbacks.forEach(clearCallback => clearCallback());
   };
+
+  function webVitalHandler(getter: (metric: Metric) => ReplayPerformanceEntry<AllPerformanceEntryData>): (data: { metric: Metric; }) => void {
+    return ({metric}) => void replay.replayPerformanceEntries.push(getter(metric));
+  }
```

Comment:
nit: let's move this up where other functions in this method are defined. Functionally it works because it gets hoisted, but it's a bit awkward to have it after the return.